redisson/src/main/java/org/redisson/api/RPermitExpirableSemaphore.java

```diff
@@ -102,20 +102,44 @@ public interface RPermitExpirableSemaphore extends RExpirable, RPermitExpirableS
     void release(String permitId);
     
     /**
-     * Returns amount of available permits.
+     * Returns number of available permits.
      *
-     * @return number of permits
+     * @return number of available permits
      */
     int availablePermits();
 
     /**
-     * Tries to set number of permits.
+     * Returns the maximum number of permits.
+     *
+     * @return maximum number of permits
+     */
+    int maximumPermits();
+
+    /**
+     * Returns the number of claimed permits.
+     *
+     * @return number of permits claimed
+     */
+    int claimedPermits();
+
+    /**
+     * Tries to set the initial number of available permits.
      *
      * @param permits - number of permits
      * @return <code>true</code> if permits has been set successfully, otherwise <code>false</code>.  
      */
     boolean trySetPermits(int permits);
-    
+
+    /**
+     * Sets the avaiable number of permits such that the number of claimed
+     * permits and the number of available permits is equal to the provided max value.
+     * If the number of permits has not been set yet, it will be set to the max value.
+     *
+     * @param permits - number of permits to use as the maximum
+     * @return the number of permits that were added
+     */
+    int setMaximumPermits(int maxPermits);
```

Comment:
Can we name it just a `setPermits()` with only difference from `trySetPermits()` that it may override previously set permits value?

Can explain why "the number of claimed permits and the number of available permits is equal to the provided max value" ?

In my vision, claimed permits are permits acquired by clients. Is that correct?

Reply:
Good catch. That comment was confusing. Yes, you're correct claimed permits is the number of permits acquired by clients. 

I've updated the function name as well to reflect that it will override the previously set number of permits.